test/integration/common.go

```diff
@@ -0,0 +1,109 @@
+//go:build integration_tests
+// +build integration_tests
+
+package integration
+
+import (
+	"bytes"
+	"regexp"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func doKprobeCheck(t *testing.T, output *bytes.Buffer) bool {
+	str := `Kprobe count: ([0-9]+)`
+	if ok, count := doProbeCommonCheck(t, output, str); ok {
+		t.Logf("counted %d application executions so far, BPF program is functioning", count)
+		return true
+	}
+	return false
+}
+
+func doAppKprobeCheck(t *testing.T, output *bytes.Buffer) bool {
+	str := `Kprobe: count: ([0-9]+)`
+	if ok, count := doProbeCommonCheck(t, output, str); ok {
+		t.Logf("counted %d application executions so far, BPF program is functioning", count)
+		return true
+	}
+	return false
+}
+
+func doProbeCommonCheck(t *testing.T, output *bytes.Buffer, str string) (bool, int) {
+	want := regexp.MustCompile(str)
+	matches := want.FindAllStringSubmatch(output.String(), -1)
+	if len(matches) >= 1 && len(matches[0]) >= 2 {
+		count, err := strconv.Atoi(matches[0][1])
+		require.NoError(t, err)
+		if count > 0 {
+			return true, count
+		}
+	}
+	return false, 0
+}
+
+func doTcCheck(t *testing.T, output *bytes.Buffer) bool {
+	if strings.Contains(output.String(), "packets received") && strings.Contains(output.String(), "bytes received") {
+		return true
```

Comment:
Add a "BPF program is functioning" log for the TC and XDP checks as is done for the other types.